### PR TITLE
Implement floor function.

### PR DESCRIPTION
**Pull request by tek1031**
This PR implements floor function with covering test for input of dtype float16, float32 and float64.

### CODE REVIEW

File: chainer/functions/math/floor.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            in_types[0].dtype.kind == 'f',
+        )
+
+    def forward(self, x):
+        xp = cuda.get_array_module(*x)
+        return utils.force_array(xp.floor(x[0]), x[0].dtype),
+
+    def backward(self, x, grad_outputs):
+        xp = cuda.get_array_module(*x)
+        return utils.force_array(xp.zeros(x[0].shape), dtype=x[0].dtype),
```
Comment (unnonouno) [inline chainer/functions/math/floor.py]: `return xp.zeros_like(x[0])` ?

File: chainer/functions/math/floor.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def floor(x):
+    """Elementwise floor function.
+
+    .. math::
+       y_i = floor(x_i)
```
Comment (unnonouno) [inline chainer/functions/math/floor.py]: `y_i = \lfloor x_i \rfloor` ?

File: chainer/functions/math/floor.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+from chainer import cuda
+from chainer import function
+from chainer import utils
+from chainer.utils import type_check
+
+
+class Floor(function.Function):
+    @property
```
Comment (unnonouno) [inline chainer/functions/math/floor.py]: Insert a blank line before this line.

**Comment by unnonouno** [comment]
Thank you for a patch. It looks good except comments.

**Comment by unnonouno** [comment]
Please fix `docs/source/reference/functions.rst`.

**Comment by tek1031** [comment]
Thank you checking my PR.
I fixed it.

File: docs/source/reference/functions.rst [hunk 4]
```diff
@@ -315,6 +315,10 @@ exp
 ~~~
 .. autofunction:: exp
 
+floor
+~~~
```
Comment (unnonouno) [inline docs/source/reference/functions.rst]: This line is too short. It needs to have the same length with "floor", i mean it requires five characters.

Comment (tek1031) [inline docs/source/reference/functions.rst]: Sorry.
I fixed.

File: chainer/functions/math/floor.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    def backward(self, x, grad_outputs):
+        xp = cuda.get_array_module(*x)
+        return xp.zeros_like(x[0]),
+
+
+def floor(x):
+    """Elementwise floor function.
+
+    .. math::
+       y_i = \lfloor x_i \rfloor
```
Comment (unnonouno) [inline chainer/functions/math/floor.py]: Two backslashes are required for each command.
`\\lfloor x_i \\rfloor`

**Comment by unnonouno** [comment]
Thank you! Great!!

**Pull request merged**